### PR TITLE
Disables several blink runtime features.

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes brave/brave-browser#11133
Fixes brave/brave-browser#11407
Fixes brave/brave-browser#11546
Fixes brave/brave-browser#11547 

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Review by AlexeyBarabash — APPROVED** [review]
LGTM

File: chromium_src/third_party/blink/renderer/core/origin_trials/origin_trial_context.cc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+// AddForceEnabledTrials only has a DCHECK with origin_trials::IsTrialValid.
+void OriginTrialContext::AddForceEnabledTrials(
+    const Vector<String>& trial_names) {
+  // When updating this list also update
+  //  * origin_trials.cc override and
+  static const char* const kBraveDisabledTrialNames[] = {
+      // [cr88] "DigitalGoods",
+      "NativeFileSystem2",
```
Comment (simonhong) [inline chromium_src/third_party/blink/renderer/core/origin_trials/origin_trial_context.cc:40]: Can we remove duplication with https://github.com/brave/brave-core/pull/6933/files#diff-dc54666ac5547a38442ebd8a02187f6df83b883532bb933408fa753a3c214868R24?

File: script/redirect-cc.py [hunk 2]
```diff
@@ -55,10 +55,13 @@ def replace_cc_arg(args):
     # Filter away out and out_x86
     # Maybe this dir can be queried from env variable instead of hardcoding
     OUT_DIR_NAMES = ['out', 'out_x86']
-    start_dir = rel_path.split(os.sep)[0]
-    if start_dir in OUT_DIR_NAMES:
-        # Don't even try to substitute path for auto-generated cc
-        return
+    rel_path_parts = rel_path.split(os.sep, 3)
+    if rel_path_parts[0] in OUT_DIR_NAMES:
+        if rel_path_parts[2] == 'gen':
```
Comment (simonhong) [inline script/redirect-cc.py:60]: 👍

**Review by simonhong — APPROVED** [review]

File: chromium_src/third_party/blink/renderer/core/origin_trials/origin_trial_context.cc (hunk 1 above)
Comment (mkarolin) [reply chromium_src/third_party/blink/renderer/core/origin_trials/origin_trial_context.cc:40]: Good point, I will rework this a bit.

**Comment by mkarolin** [comment]
Rebased.

**Comment by mkarolin** [comment]
Both **Linux** and **Android** have lint (format) errors in files that aren't related to this PR.
**iOS** has a known unit tests failure (`AdsWrapperTest testPreferencePersistance`) unrelated to this PR.

**Comment by mkarolin** [comment]
@bridiver, PTAL when you can.

File: chromium_src/third_party/blink/public/platform/disable_client_hints_browsertest.cc [hunk 3]
```diff
@@ -40,7 +42,11 @@ class ClientHintsBrowserTest : public InProcessBrowserTest {
 
   ~ClientHintsBrowserTest() override {}
 
-  void SetUp() override { InProcessBrowserTest::SetUp(); }
+  void SetUp() override {
+    // Test that even with Lang CH feature enabled, there is no header.
+    scoped_feature_list_.InitAndEnableFeature(features::kLangClientHintHeader);
```
Comment (bridiver) [inline chromium_src/third_party/blink/public/platform/disable_client_hints_browsertest.cc:47]: shouldn't we test with this both enabled and disabled?

Comment (mkarolin) [reply chromium_src/third_party/blink/public/platform/disable_client_hints_browsertest.cc]: Addressed by 296d753b902351e5d9653fecae77c58ea267253d

**Review by bridiver — APPROVED** [review]

**Comment by mkarolin** [comment]
Windows CI failure is not related to this PR (certificate issue).

**Pull request merged**